dts: nxp: Use DT_FREQ_M/K() and DT_SIZE_M/K()

butok opened the pull request:
- Use DT_FREQ_M() and DT_FREQ_K() for DTS frequency property.
- Use DT_SIZE_M() and DT_SIZE_K() for DTS size property.
- For better readability.

butok committed:
dts: nxp: Use DT_SIZE_M/K() for size

Use DT_SIZE_M() and DT_SIZE_K() for DTS size property
for better readability.

Signed-off-by: Andrej Butok <andrey.butok@nxp.com>

pdgendt commented:
Might be missing some `#include <freq.h>` or `#include <mem.h>`.

dbaluta commented on boards/nxp/mcx_n9xx_evk/mcx_n9xx_evk.dtsi:
```diff
@@ -169,11 +169,11 @@ nxp_8080_touch_panel_i2c: &flexcomm2_lpi2c2 {
 		status = "disabled";
-		size = <67108864>;
+		size = <DT_SIZE_M(64)>;
 		reg = <0>;
```
> There is no rounding down.
> 64 x 1024 x 1024 = 67,108,864

You are right. All good.

manuargue reviewed (approved):
+1 for s32

manuargue reviewed (changes requested):
in general the values look good, but I agree with above comment:

> Might be missing some `#include <freq.h>` or `#include <mem.h>`.

butok committed:
dts: nxp: Use DT_FREQ_M/K() for frequency

- Use DT_FREQ_M() and DT_FREQ_K() for DTS frequency property
for better readability.
- Include freq.h in dts to use DT_FREQ_M/K()

Signed-off-by: Andrej Butok <andrey.butok@nxp.com>
